src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -152,7 +152,10 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
   def getObjectMetadata(bucketName: String, objectKey: String, authToken: String)
                     (implicit actorRefFactory: ActorRefFactory, executionContext: ExecutionContext): Future[ObjectMetadata] = {
     val request = Get( getObjectResourceUrl(bucketName, objectKey) ) ~> addCredentials(OAuth2BearerToken(authToken)) ~> sendReceive
-    request map (_.entity.asString.parseJson.convertTo[ObjectMetadata])
+    request map { response =>
+      if(response.status.isSuccess) response.entity.asString.parseJson.convertTo[ObjectMetadata]
```

Comment:
I think unmarshal does the right thing. It does not do the json conversion if there is an error. Spray should throw and exception for an unsuccessful response code. To convert that to a different exception use recover on the Future.